src/main/kotlin/com/openlattice/chronicle/services/download/DataDownloadService.kt

```diff
@@ -143,6 +138,16 @@ class DataDownloadService(
 
     }
 
+    private fun resolveApplicationLabel(mapping: MutableMap<String, Any>): Map<String, Any> {
```

Comment:
This should be applied on writes, otherwise we will incur a pretty bad perf hit on every download unnecessarily. In addition, Kim won't have this information easily accessible and we will have two identical columns in the database (I think).

Reply:
So I figured out why application_label in android data is now the same as app_package_name

```
When an app targets Android 11 (API level 30) or higher and queries for information about the other apps that are installed on a device, the system filters this information by default. The limited package visibility reduces the number of apps that appear to be installed on a device, from your app's perspective.
```
[Package visibility](https://developer.android.com/training/package-visibility)

Android uses PackageManager class to resolve application label from app package name (https://developer.android.com/reference/android/content/pm/PackageManager)

Seems the privacy issue kicked in when we started targeting Android 11 and above, that’s why the resolution fails
It seems we can get around this by adding QUERY_ALL_PACKAGES permission in the manifest file, but using this permission is subject to Google plays policy. We would need to fill out a Declaration Form in google play
https://support.google.com/googleplay/android-developer/answer/10158779?hl=en#zippy=%2Cpermitted-uses-of-the-query-all-packages-permission

This package_name -> label resolution would help with fixing up already collected "bad" data, so we need to inject this in the data output process (downloading), but I agree we also need to do same thing on write